src/encoded/batch_download.py

```diff
@@ -309,17 +309,23 @@ def batch_download(context, request):
         for exp in results['@graph']:
             for f in exp.get('files', []):
                 if f['file_type'] in param_list['files.file_type']:
+                    if 'restricted' in f and f['restricted'] == 'true':
+                        continue
+                    else:
+                        files.append('{host_url}{href}'.format(
+                            host_url=request.host_url,
+                            href=f['href']
+                        ))
+    else:
+        for exp in results['@graph']:
+            for f in exp.get('files', []):
+                if 'restricted' in f and f['restricted'] == 'true':
+                    continue
+                else:
```

Comment:
We could replace the entire if/else block with a generator expression that gets all the experiment files and filters them based on the restriction.  Then loop through the generator, do the file_type check, and create the url.
```
    exp_files_gen = (
        exp_file
        for exp in results['@graph']
        for exp_file in exp.get('files', [])
        if not exp_file.get('restricted') == 'true'
    )
    files = []
    for exp_file in exp_files_gen:
        if exp_file['file_type'] in param_list.get('files.file_type', []):
            files.append(
                '{host_url}{href}'.format(
                    host_url=request.host_url,
                    href=exp_file['href'],
                )
            )
```
Or generate the whole thing at once
```
    files = [
        '{host_url}{href}'.format(
            host_url=request.host_url,
            href=exp_file['href'],
        )
        for exp in results['@graph']
        for exp_file in exp.get('files', [])
        if (
            not exp_file.get('restricted') == 'true' and
            exp_file['file_type'] in param_list.get('files.file_type', [])
        )
    ]
files.append
    ('{host_url}/metadata/{search_params}/metadata.tsv'.format(
        host_url=request.host_url,
        search_params=request.matchdict['search_params']
    )
)
```
I think this second one is too much.  It may be hard to update.    The first one breaks it up into meaning parts.  First we do global filters that effect all files, like 'restriction' but we could add more in the future.  Then in the generator loop we can add more specific filters like the files_type.

Also, using .get on dicts can help make conditionals.